### DISCUSSION TITLE
How is styled text handled in the source code?

### DISCUSSION THREAD

**Discussion by velllu** [helix-editor/helix#7144] [General] [answered]
If I want to edit a text to make it underlined or red, how I can do that? I tried with escape codes but they just showed like text.

**Comment by velllu** [comment]
To clarify, I mean in the actual editor source code, I mean, for example, how can I make ther bar text underlined? Or just any text

**Comment by the-mikedavis** [comment, marked as answer]
Here's an example from the source: https://github.com/helix-editor/helix/blob/d511122279b3b479362830305f4812f096273b6a/helix-term/src/ui/editor.rs#L1463-L1472

Typically the `Style` (see `helix-view/src/graphics.rs`) is something set in the theme though rather than explicit calls to `Style::fg`, `bg`, `add_modifier`, `underline_style`, etc.
